src/app/state/persist/saga.ts

```diff
@@ -211,8 +219,55 @@ function* encryptAndPersistState(action: AnyAction) {
   window.localStorage.setItem(STORAGE_FIELD, encryptedState)
 }
 
+function* retainEncryptionKeyBetweenPopupReopenings() {
+  if (runtimeIs !== 'extension') return
+  yield* fork(function* () {
+    const channelQueue = yield* actionChannel<AnyAction>('*')
+    let previousStringifiedEncryptionKey: PersistState['stringifiedEncryptionKey'] = undefined
+    while (true) {
+      yield* take(channelQueue)
+      const stringifiedEncryptionKey = yield* select(selectStringifiedEncryptionKey)
+      if (stringifiedEncryptionKey !== previousStringifiedEncryptionKey) {
+        previousStringifiedEncryptionKey = stringifiedEncryptionKey
+        yield* call(writeSharedExtMemory, stringifiedEncryptionKey)
+      }
+    }
+  })
+
+  yield* fork(function* () {
+    const encryptedState = window.localStorage.getItem(
+      STORAGE_FIELD,
+    ) as EncryptedString<PersistedRootState> | null
+    if (!encryptedState) return // Ignore
+    try {
```

Comment:
From michal: does order matter?
does look risky to write before reading the initial state

Reply:
fixed

Reply:
Hm, https://github.com/oasisprotocol/wallet/pull/2120/commits/bfe226e53c00983b6f825e84fc96df3debe5882e breaks v0  migration

Reply:
ah I lost one commit during squashing